### ISSUE TITLE
SyntaxError: Use of const in strict mode.

### ISSUE THREAD

**Issue by Morabaraba** [calebmer/postgraphql#21]
I'm on Ubuntu 14.04.3 LTS using nodejs v0.10.25. Installed with `sudo npm install -g postgraphql`. Tried to start the forum example and got the `SyntaxError: Use of const in strict mode.` error.

```sh
morabaraba@morabaraba-VirtualBox:~$ postgraphql postgres://localhost:5432/mydb --schema forum_example --development

/usr/local/lib/node_modules/postgraphql/dist/main.js:34
const manifest = JSON.parse((0, _fs.readFileSync)(_path2['default'].resolve(__
^^^^^
SyntaxError: Use of const in strict mode.
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
morabaraba@morabaraba-VirtualBox:~$ ^C
morabaraba@morabaraba-VirtualBox:~$ postgraphql --help

/usr/local/lib/node_modules/postgraphql/dist/main.js:34
const manifest = JSON.parse((0, _fs.readFileSync)(_path2['default'].resolve(__
^^^^^
SyntaxError: Use of const in strict mode.
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
morabaraba@morabaraba-VirtualBox:~$ nodejs -v
v0.10.25
```
Any suggestion or log files I can go scratch in?

**Comment by calebmer** [comment]
What version of PostGraphQL are you using? Because `1.1.1` is using strict mode in that file, [source](https://npmcdn.com/postgraphql@1.1.1/dist/main.js).

Also, is `0.10.25` a hard requirement for you? I don't plan on supporting node versions less than the `4.x` line.

**Comment by calebmer** [comment]
Closing this as I don't plan on supporting node `0.10.25`, if there's anything I can do to help though, don't be afraid to ask 😊

**Comment by Morabaraba** [comment]
Bumped node to v5.11.0. Complained about pg-connection-string, Uninstalled postgraphql, reinstalled. Installed pg-connection-string manually and it still complains.

Any suggestions?

```sh
morabaraba@morabaraba-VirtualBox:/media/sf_prj$ sudo npm install -g pg-connection-string
/usr/lib
└── pg-connection-string@0.1.3 

morabaraba@morabaraba-VirtualBox:/media/sf_prj$ postgraphql postgres://localhost:5432/mydb --schema forum_example --development
module.js:341
    throw err;
    ^

Error: Cannot find module 'pg-connection-string'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/usr/local/lib/node_modules/postgraphql/dist/main.js:18:27)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:148:18)
    at node.js:405:3
morabaraba@morabaraba-VirtualBox:/media/sf_prj$ postgraphql postgres://postgres:morabaraba@localhost:5432/mydb --schema forum_example --development
module.js:341
    throw err;
    ^

Error: Cannot find module 'pg-connection-string'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/usr/local/lib/node_modules/postgraphql/dist/main.js:18:27)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:148:18)
    at node.js:405:3
morabaraba@morabaraba-VirtualBox:/media/sf_prj$ node -v
v5.11.0
```

**Comment by calebmer** [comment]
Per-chance did you try running `npm install` twice? I've found every once in a while I'll need to do that for npm. Also, could you show me the output of `npm ls -g postgraphql` after installing?

**Comment by Morabaraba** [comment]
```
$ npm ls -g postgraphql
/usr/lib
└── postgraphql@1.1.3 
```

After executing

```sh
$ sudo npm install postgraphql -g
npm WARN deprecated lodash@1.3.1: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^4.0.0.
/usr/bin/postgraphql -> /usr/lib/node_modules/postgraphql/dist/main.js
/usr/lib
└─┬ postgraphql@1.1.3 
  ├── bluebird@3.3.5 
  ├─┬ commander@2.9.0 
  │ └── graceful-readlink@1.0.1 
  ├── dataloader@1.2.0 
  ├─┬ debug@2.2.0 
  │ └── ms@0.7.1 
  ├─┬ express@4.13.4 
  │ ├─┬ accepts@1.2.13 
  │ │ ├─┬ mime-types@2.1.10 
  │ │ │ └── mime-db@1.22.0 
  │ │ └── negotiator@0.5.3 
  │ ├── array-flatten@1.1.1 
  │ ├── content-disposition@0.5.1 
  │ ├── content-type@1.0.1 
  │ ├── cookie@0.1.5 
  │ ├── cookie-signature@1.0.6 
  │ ├── depd@1.1.0 
  │ ├── escape-html@1.0.3 
  │ ├── etag@1.7.0 
  │ ├─┬ finalhandler@0.4.1 
  │ │ └── unpipe@1.0.0 
  │ ├── fresh@0.3.0 
  │ ├── merge-descriptors@1.0.1 
  │ ├── methods@1.1.2 
  │ ├── parseurl@1.3.1 
  │ ├── path-to-regexp@0.1.7 
  │ ├─┬ proxy-addr@1.0.10 
  │ │ ├── forwarded@0.1.0 
  │ │ └── ipaddr.js@1.0.5 
  │ ├── qs@4.0.0 
  │ ├── range-parser@1.0.3 
  │ ├─┬ send@0.13.1 
  │ │ ├── destroy@1.0.4 
  │ │ ├── mime@1.3.4 
  │ │ └── statuses@1.2.1 
  │ ├── serve-static@1.10.2 
  │ ├─┬ type-is@1.6.12 
  │ │ └── media-typer@0.3.0 
  │ ├── utils-merge@1.0.0 
  │ └── vary@1.0.1 
  ├─┬ express-graphql@0.5.1 
  │ ├─┬ http-errors@1.3.1 
  │ │ └── inherits@2.0.1 
  │ └─┬ raw-body@2.1.6 
  │   ├── bytes@2.3.0 
  │   └── iconv-lite@0.4.13 
  ├─┬ graphql@0.5.0 
  │ └─┬ babel-runtime@6.6.1 
  │   └── core-js@2.3.0 
  ├── lodash@4.11.1 
  ├─┬ morgan@1.7.0 
  │ ├── basic-auth@1.0.3 
  │ └── on-headers@1.0.1 
  ├─┬ on-finished@2.3.0 
  │ └── ee-first@1.1.1 
  ├─┬ pg@4.5.5 
  │ ├── buffer-writer@1.0.1 
  │ ├── generic-pool@2.1.1 
  │ ├── packet-reader@0.2.0 
  │ ├── pg-connection-string@0.1.3 
  │ ├─┬ pg-types@1.11.0 
  │ │ ├── ap@0.2.0 
  │ │ ├── postgres-array@1.0.0 
  │ │ ├── postgres-bytea@1.0.0 
  │ │ ├── postgres-date@1.0.1 
  │ │ └─┬ postgres-interval@1.0.2 
  │ │   └── xtend@4.0.1 
  │ ├─┬ pgpass@0.0.3 
  │ │ └─┬ split@0.3.3 
  │ │   └── through@2.3.8 
  │ └── semver@4.3.6 
  ├── serve-favicon@2.3.0 
  └─┬ sql@0.67.0 
    ├── lodash@1.3.1 
    └── sliced@0.0.5 
```

**Comment by calebmer** [comment]
So it looks like `pg-connection-string` is being installed as a dependency of `pg`, but we need it to also install as a direct dependency of `postgraphql`. What version of npm are you using?

**Comment by Morabaraba** [comment]
Bumped node to `v5.11.0`.

**Comment by calebmer** [comment]
Sorry, I meant npm 😊
Does it work on `5.21`?

**Comment by calebmer** [comment]
@Morabaraba this is fixed in `1.1.5`. Turns `pg-connection-string` wasn't in PostGraphQL's `package.json` dependencies 😖

npm 3 hid this error for me because of the flat dependency installation.

**Comment by Morabaraba** [comment]
Im trying `1.2.0` with npm `3.8.6` node still on `v5.11.0`. I did a `sudo npm uninstall postgraphsql -g`.

But without any success.

```sh
morabaraba@morabaraba-VirtualBox:/media/sf_prj$ sudo npm install postgraphql -g
npm WARN deprecated lodash@1.3.1: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^4.0.0.
/usr/bin/postgraphql -> /usr/lib/node_modules/postgraphql/dist/main.js
/usr/lib
└─┬ postgraphql@1.2.0
  ├── bluebird@3.3.5
  ├─┬ commander@2.9.0
  │ └── graceful-readlink@1.0.1
  ├── dataloader@1.2.0
  ├─┬ debug@2.2.0
  │ └── ms@0.7.1
  ├─┬ express@4.13.4
  │ ├─┬ accepts@1.2.13
  │ │ ├─┬ mime-types@2.1.11
  │ │ │ └── mime-db@1.23.0
  │ │ └── negotiator@0.5.3
  │ ├── array-flatten@1.1.1
  │ ├── content-disposition@0.5.1
  │ ├── content-type@1.0.1
  │ ├── cookie@0.1.5
  │ ├── cookie-signature@1.0.6
  │ ├── depd@1.1.0
  │ ├── escape-html@1.0.3
  │ ├── etag@1.7.0
  │ ├─┬ finalhandler@0.4.1
  │ │ └── unpipe@1.0.0
  │ ├── fresh@0.3.0
  │ ├── merge-descriptors@1.0.1
  │ ├── methods@1.1.2
  │ ├── parseurl@1.3.1
  │ ├── path-to-regexp@0.1.7
  │ ├─┬ proxy-addr@1.0.10
  │ │ ├── forwarded@0.1.0
  │ │ └── ipaddr.js@1.0.5
  │ ├── qs@4.0.0
  │ ├── range-parser@1.0.3
  │ ├─┬ send@0.13.1
  │ │ ├── destroy@1.0.4
  │ │ ├── mime@1.3.4
  │ │ └── statuses@1.2.1
  │ ├── serve-static@1.10.2
  │ ├─┬ type-is@1.6.12
  │ │ └── media-typer@0.3.0
  │ ├── utils-merge@1.0.0
  │ └── vary@1.0.1
  ├─┬ express-graphql@0.5.1
  │ ├─┬ http-errors@1.3.1
  │ │ └── inherits@2.0.1
  │ └─┬ raw-body@2.1.6
│   ├── bytes@2.3.0
│   └── iconv-lite@0.4.13
├─┬ graphql@0.5.0
│ └─┬ babel-runtime@6.6.1
│   └── core-js@2.3.0
├─┬ jsonwebtoken@6.2.0
│ ├─┬ joi@6.10.1
│ │ ├── hoek@2.16.3
│ │ ├── isemail@1.2.0
│ │ ├── moment@2.13.0
│ │ └── topo@1.1.0
│ ├─┬ jws@3.1.3
│ │ ├─┬ base64url@1.0.6
│ │ │ ├─┬ concat-stream@1.4.10
│ │ │ │ ├─┬ readable-stream@1.1.14
│ │ │ │ │ ├── core-util-is@1.0.2
│ │ │ │ │ ├── isarray@0.0.1
│ │ │ │ │ └── string_decoder@0.10.31
│ │ │ │ └── typedarray@0.0.6
│ │ │ └─┬ meow@2.0.0
│ │ │   ├─┬ camelcase-keys@1.0.0
│ │ │   │ ├── camelcase@1.2.1
│ │ │   │ └── map-obj@1.0.1
│ │ │   ├─┬ indent-string@1.2.2
│ │ │   │ ├── get-stdin@4.0.1
│ │ │   │ └─┬ repeating@1.1.3
│ │ │   │   └─┬ is-finite@1.0.1
│ │ │   │     └── number-is-nan@1.0.0
│ │ │   ├── minimist@1.2.0
│ │ │   └── object-assign@1.0.0
│ │ └─┬ jwa@1.1.3
│ │   ├── buffer-equal-constant-time@1.0.1
│ │   └─┬ ecdsa-sig-formatter@1.0.5
│ │     └── base64-url@1.2.2
│ └── xtend@4.0.1
├── lodash@4.11.1
├─┬ morgan@1.7.0
│ ├── basic-auth@1.0.3
│ └── on-headers@1.0.1
├─┬ on-finished@2.3.0
│ └── ee-first@1.1.1
├─┬ pg@4.5.5
│ ├── buffer-writer@1.0.1
│ ├── generic-pool@2.1.1
│ ├── packet-reader@0.2.0
│ ├─┬ pg-types@1.11.0
│ │ ├── ap@0.2.0
   │ │ ├── postgres-array@1.0.0
   │ │ ├── postgres-bytea@1.0.0
   │ │ ├── postgres-date@1.0.1
   │ │ └── postgres-interval@1.0.2
   │ ├─┬ pgpass@0.0.3
   │ │ └─┬ split@0.3.3
   │ │   └── through@2.3.8
   │ └── semver@4.3.6
   ├── pg-connection-string@0.1.3
   ├── serve-favicon@2.3.0
   └─┬ sql@0.67.0
     ├── lodash@1.3.1
     └── sliced@0.0.5
 
 morabaraba@morabaraba-VirtualBox:/media/sf_prj$ sudo npm install postgraphql^C
 morabaraba@morabaraba-VirtualBox:/media/sf_prj$ postgraphql postgres://postgres:morabaraba@localhost:5432/mydb --schema forum_example --development
 module.js:341
     throw err;
     ^
 
 Error: Cannot find module 'pg-connection-string'
     at Function.Module._resolveFilename (module.js:339:15)
     at Function.Module._load (module.js:290:25)
     at Module.require (module.js:367:17)
     at require (internal/module.js:20:19)
     at Object.<anonymous> (/usr/local/lib/node_modules/postgraphql/dist/main.js:18:27)
     at Module._compile (module.js:413:34)
     at Object.Module._extensions..js (module.js:422:10)
     at Module.load (module.js:357:32)
     at Function.Module._load (module.js:314:12)
     at Function.Module.runMain (module.js:447:10)
     at startup (node.js:148:18)
     at node.js:405:3
```
